include/ofi_mr.h

```diff
@@ -327,6 +327,30 @@ bool ofi_mr_cache_flush(struct ofi_mr_cache *cache, bool flush_lru);
 
 int ofi_mr_cache_search(struct ofi_mr_cache *cache, const struct fi_mr_attr *attr,
 			struct ofi_mr_entry **entry);
+
+enum ofi_mr_cache_status{
```

Comment:
As a rule, do not name enums that define only flags.  As 'enum of_mr_cache_status' can't actually be used anywhere.  Just call it enum {

Reply:
Why can't it? Besides a minor space nit (no space between the name and the {), this should be fine. There is some value here for debugging. With the enum name in place, GDB would be able to identify the type without being told by the developer what flag they should be looking for. Otherwise, it becomes an undefined integer for all intents and purposes?

Reply:
Yes - all flags are integers.

When you OR flags together, the result is no longer a valid enum value.  Compilers correctly flag this as an error.  All variables *must* be declared as int and not enum ofi_mr_cache_status.  This is why the function is declared as an int, and also why ofi_mr_cache_status doesn't appear anywhere else in the patch.

Removing the name ensures that some future developer doesn't try to declare a variable as enum ofi_mr_cache_status.

The ideal solution is to change the C standard to introduce a datatype suitable for defining flags. :)

Reply:
>  Compilers correctly flag this as an error.

Which compilers?

I've used this particular behavior in the past without problems. gcc doesn't seem to have problems with this. 

```c
jswaro@uan1:~> cat test.c
#include <stdio.h>

typedef enum example_flagset {
    flag_1 = (1 << 0),
    flag_2 = (1 << 1),
    flag_3 = (1 << 2),
} example_flagset_t;

example_flagset_t test_function(void) {
    return (flag_1 | flag_3);
}

int main(int argc, char **argv) {
    example_flagset_t state;

    state = test_function();

    printf("%d\n", state);
}
```

```bash 
gcc -o tester -g test.c
```

> gdb tester

```bash
Breakpoint 1, main (argc=1, argv=0x7fffffffe378) at test.c:16
16	    state = test_function();
(gdb) print state
$1 = (unknown: 0)
(gdb) n
18	    printf("%d\n", state);
(gdb) print state
$2 = (flag_1 | flag_3)
```

Reply:
It works either way. With a typedef, or without.

Reply:
gcc is either braindead in this regard or trying to be far too clever.  Maybe there are stronger settings that could flag this?  I'm pretty sure windows and ICC are able to correctly flag (ha ha) this.  I mean, really, what's the point of declaring a variable as an enum if it can be assigned a non-enum value?